sockeye/decoder.py

```diff
@@ -148,6 +152,22 @@ def __init__(self,
                                                                      prefix="final_process_",
                                                                      num_hidden=self.config.model_size)
 
+    def state_structure(self):
+        """
+        Returns the structure of states used for manipulation of the states.
```

Comment:
I like this interface method.
Since we are relying on the string coding here, can we introduce 's', 'e', and 'd' as constants in `constants.py` and then use them here, e.g. `C.DECODER_STATE / C.ENCODER_STATE / C.INFERENCE_STATE` ?